modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/reconnection/ReconnectionTestCase.java

```diff
@@ -88,6 +89,23 @@ public void reconnectSource() throws Exception {
     });
   }
 
+  @Test
+  public void doNotStartSourceTwiceAfterExceptionOnReconnection() throws Exception {
+    ((Startable) getFlowConstruct("otherReconnectForever")).start();
+    check(5000, 1000, () -> !capturedEvents.isEmpty());
+    switchOtherFail();
+
+    checkNot(30000, 1000, () -> {
```

Comment:
I think we should make the test shorter, can we guarantee that the issue is fixed waiting less than 30 secs?

Reply:
Let me try it

Reply:
Done